lib/Onnxifi/onnxifiGlow.cpp

```diff
@@ -50,20 +57,70 @@ onnxGetBackendIDs(onnxBackendID* backendIDs, size_t* numBackends) {
 /// (onnxBackend, onnxGraph, onnxEvent) before calling this function.
 ONNXIFI_PUBLIC ONNXIFI_CHECK_RESULT onnxStatus ONNXIFI_ABI
 onnxReleaseBackendID(onnxBackendID backendID) {
+  auto *backendId = reinterpret_cast<glow::onnxifi::BackendIdPtr>(backendID);
+  if (!backendID) {
+    return ONNXIFI_STATUS_INVALID_ID;
+  }
+
+  delete backendId;
+  return ONNXIFI_STATUS_SUCCESS;
+}
+
+static onnxStatus setBackendInfoString(void *infoValue, size_t *infoValueSize,
+                                       const char *str) {
+  size_t len = strlen(str) + 1;
+  if (!infoValue || *infoValueSize < len) {
+    *infoValueSize = len;
+    return ONNXIFI_STATUS_FALLBACK;
+  }
+
+  strncpy((char *)infoValue, str, len);
+  *infoValueSize = len;
+  return ONNXIFI_STATUS_SUCCESS;
+}
+
+static onnxStatus setBackendInfoUInt64(void *infoValue, size_t *infoValueSize,
+                                       uint64_t value) {
+  if (!infoValue || *infoValueSize < sizeof(uint64_t)) {
+    *infoValueSize = sizeof(uint64_t);
+    return ONNXIFI_STATUS_FALLBACK;
+  }
+
+  *(uint64_t *)(infoValue) = value;
+  *infoValueSize = sizeof(uint64_t);
   return ONNXIFI_STATUS_SUCCESS;
 }
 
 /// Query high-level information about the backend and its target device.
-ONNXIFI_PUBLIC ONNXIFI_CHECK_RESULT onnxStatus ONNXIFI_ABI onnxGetBackendInfo(
-    onnxBackendID backendID,
-    onnxBackendInfo infoType,
-    void* infoValue,
-    size_t* infoValueSize) {
+ONNXIFI_PUBLIC ONNXIFI_CHECK_RESULT onnxStatus ONNXIFI_ABI
+onnxGetBackendInfo(onnxBackendID backendID, onnxBackendInfo infoType,
+                   void *infoValue, size_t *infoValueSize) {
   if (!infoValueSize) {
     return ONNXIFI_STATUS_INVALID_POINTER;
   }
 
-  return ONNXIFI_STATUS_SUCCESS;
+  auto *glowBackendId = reinterpret_cast<glow::onnxifi::BackendIdPtr>(backendID);
+  if (!glowBackendId) {
+    return ONNXIFI_STATUS_INVALID_POINTER;
+  }
+
+  // TODO: support more info type values. Here is the minimal required
+  // subset of info types.
+  switch (infoType) {
+  case ONNXIFI_BACKEND_NAME:
+    return setBackendInfoString(infoValue, infoValueSize, "Glow");
+  case ONNXIFI_BACKEND_VENDOR:
+    return setBackendInfoString(infoValue, infoValueSize, "Facebook");
```

Comment:
I think that the vendor should be "PyTorch", because this open source project is a part of the PyTorch group of projects.